packages/ts-morph/src/compiler/ast/common/Node.ts

```diff
@@ -1565,7 +1565,7 @@ export class Node<NodeType extends ts.Node = ts.Node> {
 
         for (const transform of transformations) {
           finalText += fileText.substring(lastPos, transform.start);
-          finalText += printer.printNode(ts.EmitHint.Unspecified, transform.compilerNode, compilerSourceFile);
+          finalText += printer.printNode(ts.EmitHint.Unspecified, transform.compilerNode, transform.compilerNode.getSourceFile());
```

Comment:
```suggestion
          finalText += printer.printNode(ts.EmitHint.Unspecified, transform.compilerNode, transform.compilerNode.getSourceFile() ?? compilerSourceFile);
```

It probably needs to be this.

Reply:
Actually, it looks like the initial code was wrong...

Reply:
No, ok I remember it now. This suggestion was correct.